### ISSUE TITLE
How to get the reasonable output from D?

### ISSUE THREAD

**Issue by wenhaoyong** [NVlabs/stylegan2-ada-pytorch#168]
when I feed the real image(from the dataset) to the discriminator, the output from it was very unreasonable. This problem has bothered me for a long time，thus hope someone can have any hints.

----------------------------------------------------------------------------------------------
the code is:
label = class_idx   // E.g class_idx=1 equal to the real class
label_dim = 10
label = torch.zeros([1, G.c_dim], device=device)
label[:, class_idx] = 1

with dnnlib.util.open_url(network_pkl) as fp:
    D = legacy.load_network_pkl(fp)['D'].requires_grad_(False).to(device) 
D = copy.deepcopy(D).eval().requires_grad_(False).to(device)
out_D = D(target_images, label)  // the shape of target_images:  torch.Size([1, 3, 32, 32])
----------------------------------------------------------------------------------------------

I expect to get a value of out_D between -10 and 10, somehow no matter I feed any real picture, the result is always a negative number <= -20.

**Comment by B0RJA** [comment]
This works in my grayscale dataset:

```
with dnnlib.util.open_url(network_pkl) as f:
	D = legacy.load_network_pkl(f)['D'].to(device)

img = PIL.Image.open("image.png").convert('RGB')
img = torch.tensor(np.array(img, dtype=np.float32)).to(device='cuda:0')
img = img[:,:,0,np.newaxis]  # RGB to gray
img = img[...,np.newaxis]
img = img.permute(3, 2, 1, 0)
img = (127.5 - img) / 127.5  # map from (0, 255) to (-1.0, 1.0)

class_idx = 0
D(img, class_idx)
```

Probably you are missing the conversion from RGB values (0-255) to the range (-1.0, 1.0).
